src/core/detail/bptree_internal.h

```diff
@@ -677,5 +685,72 @@ template <typename T> void BPTreeNode<T>::MergeFromRight(KeyT key, BPTreeNode<T>
   right->num_items_ = 0;
 }
 
+template <typename T> void BPTreePath<T>::Next() {
+  assert(depth_ > 0);
+  BPTreeNode<T>* node = Last().first;
+
+  if (node->IsLeaf()) {
```

Comment:
Can you ever start (or even be) at a non leaf position? I guess not

Reply:
yes, it can. I will add a comment.